packages/driver/src/cy/commands/request.js

```diff
@@ -277,7 +277,23 @@ module.exports = (Commands, Cypress, cy, state, config) => {
       // because we're handling timeouts ourselves
       cy.clearTimeout('http:request')
 
-      return Cypress.backend('http:request', requestOpts)
+      return Promise.try(() => {
+        // https://github.com/cypress-io/cypress/issues/6178
+        // Check if body is Blob.
+        // construct.name is added because the parent of the Blob is not the same Blob
+        // if it's generated from the test spec code.
+        if (requestOpts.body instanceof Blob || requestOpts?.body?.constructor.name === 'Blob') {
+          return Cypress.Blob.blobToBase64String(requestOpts.body).then((str) => {
+            requestOpts.body = {
+              isBinary: true,
+              base64: str,
```

Comment:
Instead of using `hasBinary` and `base64` to signal that you're sending the body as base64, let's just base64 the body and signal out-of-band in request opts, so there is never a risk of any collisions. Plus it makes the code more straightforward imo.

Something like:

```js
requestOpts.bodyIsBase64Encoded = true
requestOpts.body = str
```